### PR TITLE
docs(api): pin OpenAPI examples for the chat request bodies, default logprobs off

### PR DESCRIPTION
**Pull request by Ahmath-Gadji**
## Problem

The Swagger UI body generated for `POST /v1/chat/completions` (and `/v1/completions`) is not copy-pasteable into curl:

```json
{
  "messages": [{ "role": "user", "content": "string", "additionalProp1": {} }],
  "max_tokens": 0,
  "logprobs": true,
  "top_logprobs": 0,
  ...
}
```

Two separate causes:

1. **`additionalProp1`** — the request schemas set `extra="allow"`, which renders as `additionalProperties: true`, so Swagger UI invents a placeholder key. Extra keys are forwarded verbatim to the downstream OpenAI-compatible provider, so the generated body 400s as soon as it is used.
2. **Placeholders for unset optionals** — Swagger fills every optional with a type-default that is valid JSON but an invalid request: `max_tokens: 0` (rejected downstream, minimum is 1) and `logprobs`/`top_logprobs`, which are opt-in only.

## Changes

**Pin explicit OpenAPI examples** (`openrag/api/schemas/user/chat.py`)

`extra="allow"` has to stay — it is what forwards vendor params and `tool_calls`. So instead, each request model pins a schema-level `examples` entry via `json_schema_extra`; Swagger UI renders that in place of its own guess. The examples omit every optional that should stay omitted, so the body is copy-pasteable as-is.

- `OpenAIMessage` → `{"role": "user", "content": "What is OpenRag?"}`
- `OpenAIChatCompletionRequest` and `OpenAICompletionRequest` → model `openrag-<partition>`
- `response_format` gets an `examples=[{"type": "json_object"}]`

**Default `logprobs` to off** (chat completions only)

`logprobs` was unset by default, which left the "off" state invisible in the schema. It is now an explicit `False`.

`False` survives `model_dump(exclude_none=True)`, so the existing regression test moves its assertion one layer down to `_strip_falsy_logprobs` (`services/inference/vllm_client.py`) — which is what actually keeps an unsolicited `logprobs` off the wire, and where the guarantee belongs.

The legacy `/completions` `logprobs` is left as `None`: it is an *integer* there, where `0` is a meaningful count, so `None` remains the only correct "off" value.

## Not in scope

Two pre-existing issues surfaced while looking at the defaults, left alone here:

- `max_tokens: 0` is inconsistently handled — `_apply_default_max_tokens` treats it as an explicit client value (`is not None`) while `validate_tokens_limit` treats it as falsy (`or default`). The two layers disagree and `0` is forwarded, then rejected downstream. No `ge=1` bound exists.
- `top_logprobs`, `temperature` and `top_p` are unbounded.

Both need a strict-vs-permissive call on how far to validate client input, so they are worth their own PR.

## Testing

- `uv run pytest tests/unit/api tests/unit/services/inference` → 747 passed
- Verified in the generated OpenAPI document that `logprobs` has `default: false` and that the pinned examples land under `components.schemas.<Model>.examples`

## Summary by CodeRabbit

* **New Features**
  * Added copy-ready API request examples, including supported response format values.
  * Swagger UI now displays accurate request bodies instead of placeholder keys or invalid values.

* **Bug Fixes**
  * Chat requests now default `logprobs` to `False`.
  * Falsy log-probability settings are omitted before requests are sent to providers.
  * Preserved valid log-probability settings when they are explicitly enabled.
  * Added validation to ensure request examples and log-probability behavior remain consistent.

### CODE REVIEW

File: openrag/api/schemas/user/chat.py [hunk 1]
```diff
@@ -16,8 +26,32 @@ class OpenAIMessage(BaseModel):
...
+    # requests -- `max_tokens: 0` (rejected downstream, minimum is 1) and
+    # `logprobs: true` / `top_logprobs: 0` (opt-in only). The example
+    # omits every field that should stay omitted, so it is copy-pasteable.
+    model_config = ConfigDict(
+        extra="allow",
+        json_schema_extra={
+            "examples": [
+                {
+                    "model": "openrag-<partition>",
+                    "messages": [{"role": "user", "content": "What is OpenRag?"}],
```
[the remark this answers is not in the document]
Comment (Ahmath-Gadji) [reply openrag/api/schemas/user/chat.py]: Correct that `openrag-<partition>` returns 404 if used verbatim — `get_partition_name()` resolves the partition to the literal `<partition>`, which does not exist.

This is deliberate. It is a fill-in-the-blank placeholder: the reader substitutes their own partition name before sending. The alternative, `openrag-all`, does work out of the box but hides the normal usage (a request is scoped to a partition) and only resolves for a user who already has memberships.

The problem this PR set out to fix was Swagger UI generating `"additionalProp1": {}` — a key that looks valid and silently 400s downstream. `<partition>` is not that: it is obviously a slot to fill.

Not changing.

File: openrag/api/schemas/user/chat.py [hunk 2]
```diff
@@ -31,14 +65,11 @@ class OpenAIChatCompletionRequest(BaseModel):
     # more. The router fills it from the resolved endpoint via
     # ``_apply_default_max_tokens`` once the partition is known.
     max_tokens: int | None = Field(None)
-    # Client-controlled and forwarded as-is to the downstream model; the server
-    # default is off (see LLMParamsConfig.logprobs). For chat completions
-    # `logprobs` is a boolean — `top_logprobs` carries the count — unlike the
-    # legacy /completions endpoint where `logprobs` is an integer.
-    logprobs: bool | None = Field(None)
+    logprobs: bool | None = Field(False)
```
[the remark this answers is not in the document]
Comment (Ahmath-Gadji) [reply openrag/api/schemas/user/chat.py:74]: Real gap, fixed in #899 — but not caused by this PR.

`LLMParamsConfig.logprobs` already defaults to `False`, so it lands in `OllamaClient._defaults` and is merged into every request body regardless of what the client sends. Reproduced on `develop` with a mock transport and **no** client-supplied `logprobs` at all:

```
{"logprobs": false, "temperature": 0.1, "model": "m", "messages": [...], "stream": false}
```

Defaulting `logprobs` to `False` in the schema changes which dict the `false` arrives in, not whether it is sent. So the fix is orthogonal to this PR and is split out into #899, which wires the existing `_strip_falsy_logprobs` into `chat`, `stream_chat` and `generate` (4 lines — `ollama_client` already imports from `vllm_client`), with 6 tests covering both directions, including that integer `logprobs: 0` on `/v1/completions` survives.

**Comment by hedhoud** [comment]
The change looks correct and all CI checks are green. One minor test gap remains: the tests do not verify the generated OpenAPI output itself, including that the explicit examples appear and Swagger no longer emits the additionalProp1 placeholder. This does not indicate a runtime defect, so I am approving the PR.

**Review by hedhoud — DISMISSED** [review]
Reviewed the request schemas, serialization flow, edge cases, and CI results. No blocking issue found; the remaining OpenAPI example assertion is a minor test-coverage improvement.

**Review by hedhoud — APPROVED** [review]
